extensions/azurePublishNew/src/recoilModel/atoms/resourceConfigurationState.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { atom } from 'recoil';
+
+import { tenantIdSelector } from '../selectors/tenants';
+import { UserInfo } from '../types';
+
+export const defaultUserInfo: UserInfo = {
+  email: undefined,
+  expiration: undefined,
+  name: undefined,
+  sessionExpired: undefined,
+  token: undefined,
+};
+
+export const tenantSelectionState = atom<string>({
+  key: 'resourceConfiguration_tenant',
+  default: tenantIdSelector,
```

Comment:
I'm confused by this, can you just use the selector itself and add a setter for it?

Reply:
Changed it to a dispatcher.